crates/js-sys/src/lib.rs

```diff
@@ -463,6 +503,152 @@ extern "C" {
     pub fn values(this: &Array) -> Iterator;
 }
 
+/// The `Atomics` object provides atomic operations as static methods.
+/// They are used with `SharedArrayBuffer` objects.
+///
+/// The Atomic operations are installed on an `Atomics` module. Unlike
+/// the other global objects, Atomics is not a constructor. You cannot
+/// use it with a new operator or invoke the `Atomics` object as a
+/// function. All properties and methods of `Atomics` are static
+/// (as is the case with the Math object, for example).
+/// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics)
+#[allow(non_snake_case)]
+pub mod Atomics {
+    use super::*;
+
+    #[wasm_bindgen]
+    extern "C" {
+        #[derive(Clone, Debug)]
+        #[wasm_bindgen(extends = Object)]
+        pub type Atomics;
+
+        /// The static `Atomics.add()` method adds a given value at a given
+        /// position in the array and returns the old value at that position.
+        /// This atomic operation guarantees that no other write happens
+        /// until the modified value is written back.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/add)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn add(typed_array: &JsValue, index: u32, value: f64) -> Result<f64, JsValue>;
```

Comment:
@alexcrichton since it returns the old value, is it fine to use `Result<f64, JsValue>` as a function return argument?
Another question: `typed_array` argument can be of any type, for example `Int8Array`. Is it fine to keep `value` type as `f64`?

Reply:
Yeah I think `f64` is appropriate here since it's largely just "some JS number" which becomes an f64 anyway. Additionally returning `Result<f64, JsValue>` sounds good to me as well!